app/src/main/java/com/example/encoratask/model/Character.kt

```diff
@@ -0,0 +1,46 @@
+package com.example.encoratask.model
+
+import org.json.JSONObject
+import java.io.Serializable
+
+/**
+ * Created by Erick Sanchez
+ * Revision 1 - 09/06/21
+ */
+class Character:Serializable {
+
+    var id:Int = 0
+    var name:String = ""
+    var status = ""
+    var species = ""
+    var gender = ""
+    var image:String = ""
+
+    constructor(obj:JSONObject) {
+
+        this.id = obj.optInt("id")
+        this.name = obj.optString("name")
+        this.status = obj.optString("status")
+        this.species = obj.optString("species")
+        this.gender = obj.optString("gender")
+        this.image = obj.optString("image")
+    }
+
+    companion object {
+
+        fun parse(response:String) : ArrayList<Character> {
+
+            var characters = ArrayList<Character>()
+            val obj = JSONObject(response)
+            val results = obj.optJSONArray("results")
+            if (results != null) {
+
+                for (i in 0 until results.length()) {
```

Comment:
If you use Gson library you could avoid the iteration of the object results and create directly a list of Character

Reply:
Sounds good, but I thnik use 3rd party libraris to solve this easy things its bad idea, and GSON has some bad popularity

https://medium.com/@dannydamsky99/heres-why-you-probably-shouldn-t-be-using-the-gson-library-in-2018-4bed5698b78b

https://medium.com/@dannydamsky99/heres-why-you-probably-shouldn-t-be-using-the-gson-library-in-2018-4bed5698b78b

For that reason I choosed make it manually, sorry =(

Reply:
i haven't had any issues using GSON since its a proven library with active development, remember that we don't need to reinvent the wheel

Reply:
BTW for kotlin there's another parser that will better process null-safety but i forgot its name 🤔